### ISSUE TITLE
`start_vertices` is unfriendly to laziness

### ISSUE THREAD

**Issue by leshabirukov** [HeWeMel/nographs#25]
Hello again! Test based on example, familiar for you:
Setup:
```
  import nographs as nog
  
  def next_edges(i, _):
      j = (i + i // 6) % 6
      yield i + 1, j * 2 + 1
      if i % 2 == 0:
          yield i + 6, 7 - j
      elif i % 1200000 > 5:
          yield i - 6, 1
  
  traversal = nog.TraversalShortestPaths(next_edges)
```
Now if start_vertices is lazy:
```
  traversal.start_from( start_vertices=(      map(lambda x:x+1, (-1,) )), build_paths=True).go_to(5)
```
Gives `KeyError: 'Vertex not found, graph exhausted.'`
Kicked with `list`
```
  traversal.start_from( start_vertices=( list ( map(lambda x:x+1, (-1,) ))), build_paths=True).go_to(5)
```
Do fine. 
Not a big deal actualy, but laziness support here would be good.

**Comment by leshabirukov** [comment]
> 2\. As far as I remember, there is no collection.something that can be used to specify this property (without additional effects).

`collections.abc.Container`? it says 'True' on list, tuple and range, while 'False' on map and generator.

Another thing, may be it is possible to use `itertools.tee` to uniformly create iterators for list-likes and lazy-s?

**Comment by HeWeMel** [comment]
Problem solved in v3.4.0.

See [ChanceLog](https://nographs.readthedocs.io/en/latest/changelog.html#:~:text=Methods%20start_from%20of%20traversals%3A%20Argument%20for%20parameter%20start_vertices%20is%20traversed%20at%20most%20once.%20Thus%2C%20it%20can%20be%20an%20Iterator%20or%20a%20Generator)
